### DISCUSSION TITLE
Can't sync

### DISCUSSION THREAD

**Discussion by elena87** [audiamus/BookLibConnect#40] [General]
Hi, I created a profile (I can see it in my Amazon account under My devices), I have some audiobooks in my Audible library, but in the Book Lib software they don't show up, it's empty. What I have to do? Thank you.

**Comment by audiamus** [comment]
The log files will tell more. And the Audible server response json files. 

Log files: `C:\Users\<User>\AppData\Local\audiamus\BookLibConnect\log`

Response files: `C:\Users\<User>\AppData\Local\audiamus\BookLibConnect\tmp`

The response files are json format text files, but in compacted form, the way Audible sends them. Some text editors like [Visual Studio Code](https://code.visualstudio.com/) can reformat them to make them readable. The `LibraryResponse` files are of particular interest as they list the books for your account.

**Comment by elena87** [comment]
In the LibraryResponse there is in fact the list of my Audible account. 
Here's the log instead.
The app is still blank.
[BookLibConnect_2022-05-09_001.log](https://github.com/audiamus/BookLibConnect/files/8649774/BookLibConnect_2022-05-09_001.log)

**Comment by audiamus** [comment]
Yes, there are books listed in the log. I would need more details about those books and therefore the file "LibraryResponse-2022_05_09-09_46_26.json" from the temp folder. You can send it to me as a personal message attachment via reddit.com. My user name there is also "audiamus".

**Comment by saifer92** [comment]
Hi, I've the same problem, no book listed in app, but in my case JSON file shows item array empty...
Here my log (I've censored my account alias, just in case... 😄 ) and LibraryResponse (addex .txt to be uploaded here)

[LibraryResponse-2022_09_05-20_20_24.json.txt](https://github.com/audiamus/BookLibConnect/files/9491594/LibraryResponse-2022_09_05-20_20_24.json.txt)
[BookLibConnect_2022-09-05_005.log](https://github.com/audiamus/BookLibConnect/files/9491593/BookLibConnect_2022-09-05_005.log)

Thank you!

**Comment by audiamus** [comment]
> in my case JSON file shows item array empty

The log says the same, the server response is empty, no books (i.e. meta info fror them) are returned. 

Did you try a full resync? With a resync, the date filter "since last update" will be disabled. 

If still empty, are you sure that you connected to the right profile for your books: the combination of the Amazon/Audible account plus the region?

_(The previous case in this discussion turned out to be about podcasts which are not yet supported.)_

**Reply by saifer92** [reply]
Yes, it was my case, too. There were only podcasts in my library, by adding a book it's now working. Thank you!

**Comment by rdmcq** [comment]
BookLib not listing books in my library. Tmp dir contains only RegistrationReponse files (no LibraryResponse files).
[BookLibConnect_2023-03-09_004.log](https://github.com/audiamus/BookLibConnect/files/10937664/BookLibConnect_2023-03-09_004.log)

**Reply by audiamus** [reply]
From what I see in the log file, you worked with two profiles, aliases "Account 1" and "Account 2". Apparently, creating "Account 1" didn't really go through, as something fundamentally went wrong when connecting to the Audible servers: "No such host is known". 

The "Account 2" profile then registered successfully. However, with the "Account 2" profile, the Audible Servers were not queried for your library. All the `[AudibleApi.GetLibraryAsync]` entries in the log are for "Account 1", where the profile setup was not completed.

You can try start Book Lib Connect again, select the "Account 2" profile, open the [Download] window and [Resync] the library. That should trigger the GetLibrary() calls.

**Comment by iSams123** [comment]
Hello, I got the same issue where my Audible library has 16 titles but when using BookLib to sync the profile for download, it only displays 15. I tried resync but the problem still persisted. The audiobook belongs to a serie among other audiobooks so I'm not sure why it does not show.

I inspected the log file and see  #books=16 and the LibraryResponse also has the name of the missing audiobook ("Walzer in Wien"). However, it didnt show in the software.

[BookLibConnect_2023-04-05_004.log](https://github.com/audiamus/BookLibConnect/files/11158846/BookLibConnect_2023-04-05_004.log)
[LibraryResponse-2023_04_05-13_36_38.json.txt](https://github.com/audiamus/BookLibConnect/files/11158856/LibraryResponse-2023_04_05-13_36_38.json.txt)
![image](https://user-images.githubusercontent.com/13537361/230084137-43eccb70-7a71-461c-9113-c8885e1d054e.png)

**Reply by audiamus** [reply]
The json file (_thank you_) says that Audible marked "Walzer in Wien" as an "adult" title. Therefore, BookLibConnect does not show it by default. Enable "adult products" in the main window.

**Comment by nurbles** [comment]
Sometime between my log from 2023-03-16 and 2023-04-28 something must've changed in the AudibleApi because, starting with the 04-28 log, all of the AudibleApi sessions look like this:
```
15:16:12.758 0001 [AudibleApi.GetLibraryAsync] >>>
15:16:12.848 0001 [AudibleApi.sendForStringAsync] HttpRequestException: "Response status code does not indicate success: 403 (Forbidden)."
{"message":"Request could not be authenticated"}
15:16:12.848 0001 [AudibleApi.GetLibraryAsync] <<< 
15:16:12.848 0001 [AudibleApi.DownloadCoverImagesAsync] >>> 
```
and nothing I've purchased since 03-16 has ever been added to my library in BookLibConnect when I try to sync.  I can still manually download and use AaxAudioConverter to make the MP3s I use for books in my car, but it would be nice it this still worked.  Or is Audible "fighting back" by changing things to make your work more difficult?

If there's anything I can provide to help figure this, tests I can run for you, etc.  Please let me know.

**Reply by audiamus** [reply]
I don't think this is down to some API change, it looks more like a mismatch between access token, registered device and account. Have you changed anything in your environment recently?

**Reply by nurbles** [reply]
Other than Windows 10 updating itself, I don't think so.  I even left the Audible Manager installed, just in case it was needed.  That said, I suppose a maintenance tool could've "cleaned" something belonging to BookLibConnect out of the registry or a temporary files folder without me realizing.  Also, strangely, it wouldn't sync the three books I purchased _prior_ to 03-16, either -- only the book I bought **on** 03-16-23.  That _REALLY_ confuses me.  Let me know if there are any tests I can run, configuration settings I can clean/reset or whatever and I'll be happy to give them a try.

**Reply by audiamus** [reply]
You could try to register anew. This 403 error looks very much like an authentication issue.

**Reply by nurbles** [reply]
I didn't find any "register" option, but I did create a new profile for my account (which apparently replaced the previous one) and now it is working.  I was forced to change my Amazon/Audible password a while back and that must have been the problem.  I don't remember ever needing to do that with the Audible Manager, but it was kinda junky -- yours is far superior.  Sorry for my confusion.

**Reply by audiamus** [reply]
Well, yes, "new profile" is the way to re-register, sorry. And yes, this makes sense. Changing your password could very well invalidate your access token. A security measure by Amazon to protect your account.

**Comment by Mosai57** [comment]
I'm encountering this same issue. Forcing a full sync doesn't load the books in my library. I've removed and readded my profile on both the local BookLib end as well as on my Amazon devices. Any advice would be appreciated!
[BookLibConnect_2023-08-09_008.log](https://github.com/audiamus/BookLibConnect/files/12307031/BookLibConnect_2023-08-09_008.log)
[LibraryResponse-2023_08_09-13_55_00.json.txt](https://github.com/audiamus/BookLibConnect/files/12307035/LibraryResponse-2023_08_09-13_55_00.json.txt)

**Reply by audiamus** [reply]
Somewhat strange. It appears you can successfully download your library meta data and the full resync also seems to work. However, the log suggests that none of these books is added to the local database. This may have to do with account management. Did you use a different account with Book Lib Connect before the current one? There could be an issue with account keys in the database,

To make sure, I suggest to remove the database and start afresh. 

To do that, stop Book Lib Connect, then navigate to `C:\Users\<USERNAME>\AppData\Local\audiamus\BookLibConnect\data`. Move or rename `audiobooks.db`. Your account info will not be affected, it's not stored in the database. Start Book Lib Connect. It will create and fill a new database. You will be asked to associate your Audible account again, but that's just for the local database.

**Reply by Mosai57** [reply]
Looks like that was the fix!

**Reply by audiamus** [reply]
Very good. I put this on my todo list, it shouldn't happen.

**Reply by nurbles** [reply]
Renaming the file worked for me, too.  Interestingly (to me, at least) is that the NEW file is 672 KB while the OLD file was 1032 KB.  Apparently the format changed to something new and more compact?  I'm certain that I've never REMOVED any books from my library, and the old library did not even show all the books I did have.  So why is the fixed one (that includes MORE entries) so much smaller?

Not really important, just curious.  (smile)  Thanks for the 'fix'!

**EDIT**  Thanks for the explanation, too!

**Reply by audiamus** [reply]
SQLite database file size is influenced by usage history. A newly created and populated database will usually be more compact than one that has been populated over time and also may have undergone changes in table definitions. All for performance reasons. SQLite tools offer functionality to explicitly compact the database. As the Book Lib Connect database is fairly small, compacting it once in a while does not seem to be necessary.
